tools/gen-magic-strings.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import configparser
+import json
+import os
+import re
+
+from settings import PROJECT_DIR
+
+
+def read_magic_string_definitions():
+    ini_parser = configparser.ConfigParser()
+    ini_parser.optionxform = str # case sensitive options (magic string IDs)
+    ini_parser.read(os.path.join(PROJECT_DIR,
+                                 'jerry-core',
+                                 'lit',
+                                 'lit-magic-strings.ini'))
+
+    defs = [(str_ref, json.loads(str_value))
+            for str_ref, str_value in ini_parser.items('LIT_MAGIC_STRINGS')]
+    defs = sorted(sorted(defs, key=lambda ref_value: ref_value[1]),
+                  key=lambda ref_value: len(ref_value[1]))
+    return defs
+
+
+def extract_magic_string_references():
+    results = dict()
+
+    def process_line(fname, lnum, line, guard_stack):
+        for str_ref in re.findall('LIT_MAGIC_STRING_[a-zA-Z0-9_]+', line):
+            if str_ref in ['LIT_MAGIC_STRING_DEF',
+                           'LIT_MAGIC_STRING_FIRST_STRING_WITH_SIZE',
+                           'LIT_MAGIC_STRING_LENGTH_LIMIT',
+                           'LIT_MAGIC_STRING__COUNT']:
+                continue
+
+            guard_set = set()
+            for guards in guard_stack:
+                for guard in guards:
+                    guard_set.add(guard)
+            guard_tuple = tuple(sorted(guard_set))
+
+            if str_ref not in results:
+                results[str_ref] = dict()
+            str_guards = results[str_ref]
+
+            if guard_tuple not in str_guards:
+                str_guards[guard_tuple] = list()
+            file_list = str_guards[guard_tuple]
+
+            file_list.append((fname, lnum))
```

Comment:
Right now, we aren't. This script started off as a checker, not as a generator, and in that phase it was mandatory/useful to be able to print out locations of violations (not just file names but line numbers as well). Then it turned out that checking/proving correctness is harder that generating the right thing, that's when this script became a generator. Now, line numbers are not that needed, but are still useful for debugging (the script itself), so I've kept them.